backend/backend-standard-and-best-practices.md

```diff
@@ -0,0 +1,278 @@
+      "email": "My email"
+    },
+    {
+      "id": 2,
+      "first_name": "My first name",
+      "last_name": "My last name",
+      "email": "My email"
+    },
+    {
+      "id": 3,
+      "first_name": "My first name",
+      "last_name": "My last name",
+      "email": "My email"
+    }
+  ]
+}
+```
+### 2.2- Output parameters structure
+
+If we need to return an object that represents a certain entity, for example an user, we should avoid to return it using a key in the body containing all its data; we should directly return the data in the body instead.
+
+```json
+{
+  "id": 1,
+  "first_name": "My first name",
+  "last_name": "My last name",
+  "email": "My email"
+}
+```
+
+instead of:
+
+```json
+{
+  "user": {
+    "id": 1,
+    "first_name": "My first name",
+    "last_name": "My last name",
+    "email": "My email"
+  }
+}
+```
+
+## 3- Response bodies
+
+### 3.1- Pagination
+
+All paginated lists requested to our APIs should be returned using the following structure:
+
+```json
+{
+ "page": [
+    {
+      "id": 4,
+      "first_name": "My first name",
+      "last_name": "My last name",
+      "email": "My email"
+    },
+    {
+      "id": 5,
+      "first_name": "My first name",
+      "last_name": "My last name",
+      "email": "My email"
+    },
+    {
+      "id": 6,
+      "first_name": "My first name",
+      "last_name": "My last name",
+      "email": "My email"
+    }
+ ],
+ "count": 3,
+ "limit": 3,
+ "offset": 0,
+ "total_pages": 4,
+ "total_count": 12,
+ "previous_page": 1,
+ "current_page": 2,
+ "next_page": 3,
+ "previous_page_link": "domain.com/users?page=1&limit=3",
+ "current_page_link": "domain/com/users?page=2&limit=3",
+ "next_page_link": "domain.com/users?page=3&limit=3",
+ "first_page_link": "domain.com/users?page=1&limit=3",
+ "last_page_link": "domain.com/users?page=4&limit=3",
+ "n_page_link": "domain.com/users?page=X&limit=3"
+}
+```
+
+Some clarifications about the pagination response:
+- The starting page number should always be 1.
+- When returning the first page, the *previous_page* property should be `NULL`.
+- The parameters with the `_link` suffix are suggested when using link-oriented pagination; otherwise, they should not be included in the response body.
+
+### 3.2- Error forwarding
+
+When informing a client of a request error, the response body should be returned using the following structure:
+
+```json
+{
+  "errors": [
+    { "code": "111", "message": "age must be an int" }
+    { "code": "112", "message": "email is mandatory" }
+  ],
+  "origin": "api-name",
+  "stack_trace": "...",
+  "timestamp": "2019-09-10 03:14:07"
+}
+```
+
+Some clarifications about the error response:
+- The `timestamp` field is not mandatory, but its use is recommended for logging purposes.
+- Both the `origin` and `stack_trace` fields are considered sensitive information, and they should not be sent to a frontend or external client. Their use should be limited between internal APIs of the same application.
```

Comment:
This fields could be scoped by environment too, in development at least seems useful